salt/utils/dns.py

```diff
@@ -1,18 +1,787 @@
+https://blog.widodh.nl/2016/07/calculating-ds-record-from-dnskey-with-python-3/
+https://github.com/saltstack/salt/pull/36180
+- modules:
+    - ddns
+    - dnsutil
+    - libcloud dns
+- states
+    - libcoud dns
+- utils
+    - network
+
+# should fix
+https://github.com/saltstack/salt/issues/25953
+https://github.com/saltstack/salt/issues/20275
+'''
+from __future__ import print_function, absolute_import
+# Python
+import base64
+import binascii
+import hashlib
 import itertools
+import socket
+import ssl
+import string
+from salt.ext.six.moves import zip
+from salt._compat import ipaddress
+from salt.utils.odict import OrderedDict
+
+# Salt
+import salt.modules.cmdmod
+import salt.utils
+
+# Debug & Logging
 import logging
+import pprint
 
-# Import salt libs
-import salt.utils.network
-from salt._compat import ipaddress
+# Integrations
+try:
+    import dns.resolver
+    HAS_PYDNS = True
+except ImportError as e:
+    HAS_PYDNS = False
+HAS_DIG = salt.utils.which('dig') is not None
+HAS_DRILL = salt.utils.which('drill') is not None
+HAS_HOST = salt.utils.which('host') is not None
+HAS_NSLOOKUP = salt.utils.which('nslookup') is not None
 
+__salt__ = {
+    'cmd.run_all': salt.modules.cmdmod.run_all
+}
 log = logging.getLogger(__name__)
+ppr = pprint.PrettyPrinter(indent=2).pprint
+
+
+class RFC(object):
+    # http://www.iana.org/assignments/dns-sshfp-rr-parameters/dns-sshfp-rr-parameters.xhtml
+    SSHFP_ALGO = OrderedDict((
+        (1, 'rsa'),
+        (2, 'dsa'),
+        (3, 'ecdsa'),
+        (4, 'ed25519'),
+    ))
+
+    SSHFP_HASH = OrderedDict((
+        (1, 'sha1'),
+        (2, 'sha256'),
+    ))
+
+    # http://www.iana.org/assignments/dane-parameters/dane-parameters.xhtml
+    TLSA_USAGE = OrderedDict((
+        (0, 'pkixta'),
+        (1, 'pkixee'),
+        (2, 'daneta'),
+        (3, 'daneee'),
+    ))
+
+    TLSA_SELECT = OrderedDict((
+        (0, 'pkixta'),
+        (1, 'pkixee'),
+        (2, 'daneta'),
+        (3, 'daneee'),
+    ))
+
+    TLSA_MATCHING = OrderedDict((
+        (0, 'full'),
+        (1, 'sha256'),
+        (2, 'sha512'),
+    ))
+
+    SRV_PROTO = (
+        'tcp',
+        'udp',
+        'sctp'
+    )
+
+    @staticmethod
+    def validate(lookup, ref, match=None):
+        if lookup in ref:
+            return lookup
+        elif match == 'in':
+            return [code for code, name in ref.items() if lookup in name][-1]
+        else:
+            # OrderedDicts only!(?)
+            return ref.keys()[ref.values().index(lookup)]
+
+
+def _data2rec(rschema, rdata):
+    '''
+    OrderedDict({
+        'prio': int,
+        'weight': int,
+        'port': to_port,
+        'name': str,
+    })
+    '10 20 25 myawesome.nl'
+
+    '''
+    try:
+        rdata = rdata.split(' ', len(rschema))
+        rschema = rschema.items()
+        return dict((
+            (fname, rcb(rdata)) for (fname, rcb), rdata in zip(rschema, rdata)
+        ))
+    except:  # pylint: disable=bare-except
```

Comment:
Please don't disable this lint check, it's there for a reason. Remove the pylint disable and use ``except Exception`` instead. A bare except catches things like ``SystemExit`` and other exceptions we want to be caught elsewhere. I'm not saying that this code would raise a ``SystemExit``, but bare exceptions are something we try to avoid for good reason.